src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1936,12 +1936,9 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Receive(uint8_t aChannel
 
     VerifyOrExit(mState != kStateDisabled, error = OT_ERROR_INVALID_STATE);
 
-    if (mChannel != aChannel)
-    {
-        error = Set(SPINEL_PROP_PHY_CHAN, SPINEL_DATATYPE_UINT8_S, aChannel);
-        SuccessOrExit(error);
-        mChannel = aChannel;
-    }
+    error = Set(SPINEL_PROP_PHY_CHAN, SPINEL_DATATYPE_UINT8_S, aChannel);
+    SuccessOrExit(error);
+    mChannel = aChannel;
```

Comment:
Is it workable to cache the the transmitting channel in mChannel after a frame TX request?

Reply:
This can also work. Was considering it as well. We may need to do the same on energy scan.

Reply:
I think we should avoid unnecessary calls to Set(SPINEL_PROP_PHY_CHAN, ...) since it's a blocking operation, thus having significant cost.

Reply:
That's exactly the reason I was thinking of the same. It is already changed in new push.